homework/Task 2/DiagonalMatrix.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Runtime.Remoting.Messaging;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace homework._Task2
+{
+    public class DiagonalMatrix
+    {
+        private readonly int _size;
+        private int[] _diagonal;
+        public DiagonalMatrix(int[] diagonal) 
+        {
+            if (diagonal != null)
+            {
+                _size = diagonal.Length;
+                _diagonal = diagonal;
+            }
+        }
+
+        public int Track()
+        {
+            int sum = 0;
+            foreach(int value in _diagonal)
+            {
+                sum += value;
+            }
+            return sum;
+        }
+
+        public int GetSize()
+        {
+            return _size;
+        }
+
```

Comment:
This is not indexer, please redo.
https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/indexers/